doc/release-notes.md

```diff
@@ -86,6 +86,8 @@ Low-level Changes section below.
 GUI changes
 -----------
 
+- The "Start Bitcoin Core on system login" option has been removed on macOS.
```

Comment:
Please mention that it was already not working on recent MacOS versions.

Reply:
This change will actually be user facing, as the current release binaries include this option in the preferences menu (0.19 was compiled with a macOS 10.10 minimum and targeting the 10.11 SDK).

Reply:
So the option was there, but it wasn't working on newer MacOS?

Reply:
The option was there, and it should have been working for all versions of macOS (I can only test up to 10.14). Note that there are plenty of ways for the user to create the "start at login" behaviour without us providing an option from within Bitcoin Core.

Reply:
Ok, thanks for explaining.

The way MacOS does things is really confusing here. So we need to remove the functionality because the calls are deprecated, but they still work fine, even on newer MacOS, if compiled with an older MacOS in mind.

> Note that there are plenty of ways for the user to create the "start at login" behaviour without us providing an option from within Bitcoin Core.

I remember we used a different way based on scripts in the past. That way was deprecated, too :angry: Although annoying for users which rely on this functionality, I tend to agree it's not worth coming up with a new hack here every time.

Reply:
> they still work fine, even on newer MacOS, if compiled with an older MacOS in mind
They work just fine even when compiling for the latest macOS